### DISCUSSION TITLE
Where to store assets for a Quarto website?

### DISCUSSION THREAD

**Discussion by maelle** [quarto-dev/quarto-cli#6876] [Q&A] [answered]
### Description

Imagine I have font files and images that I want to store in my website source and publish at the root, or in a subfolder, of the rendered website. Where would I store those?

In a Hugo website, that would correspond to the static/ folder: if I have `static/img/logo.png` in my website `https://example.com` source, the logo will be at `https://example.com/img/logo.png`. I'm looking for the equivalent Quarto concept.

**Comment by hrbrmstr** [comment, marked as answer]
For a recent site I stood up, I did:

```
resources:
  - "data/*"
```

In `_quarto.yml`.

You can add as many dirs as you like and can reference the files in `data` or wherever with that dir prefix.

In: https://companion.hrbrmstr.dev/posts/2023-09-16-rbql/ , the data file is read from `https://companion.hrbrmstr.dev/data/mtcars.csv`.

I'd link to the repo for the site, but it's not on any social coding service.

h/t to Perplexity: https://www.perplexity.ai/search/in-a-quarto-O4GCx4BnQl.lzAeCFddmyA?s=c b/c I could not find out this info after a quick docs scan.

**Reply by maelle** [reply]
Thank you!!

**Reply by patgd** [reply]
> h/t to Perplexity … b/c I could not find out this info after a quick docs scan.

I did not know about this. Thank you for sharing.

**Comment by rgaiacs** [comment]
From https://quarto.org/docs/websites/website-tools.html#site-resources

> Quarto will automatically detect any files that you reference within your site and copy them to the output directory

If the auto-detect fail, you can use `resources` in your `_quarto.yml`

```
project:
  type: website
  resources: 
    - "*.xlsx"
```

or in the YAML header

```
---
title: "My Page"
resources:
  - "sheet.xlsx"
---

Lorem ipsum
```

If you are working in a Shiny project, check https://quarto.org/docs/interactive/shiny/resources.html#asset-resources.

**Reply by maelle** [reply]
Thank you!!

**Reply by AleksandarMedarevic** [reply]
Hi,

```
project:
  type: book
  resources: 
    - "*.xlsx"
```

On the page I wrote:

`[PDF obrazac](/documents/Upitnik_za_popunjavanje_(samo_za_internu_primenu_u_ustanovi).pdf)`

Quartro created a subfolder  _documents_ within _book folder. However, when I uploaded that page and the documents folder in my GitHub repo, link is not walid. 

_File not found

The site configured at this address does not contain the requested file._

How to fix this issue?

**Reply by mcanouil** [reply]
Quarto only uses resources inside the project. A leading `/` means root of the Quarto project as documented.
Also, having parentheses in a file name is really a bad idea in general.

PS: next time, please open a separate discussion as your issue is off topic here.

**Comment by mcanouil** [comment]
I usually put the assets into `/assets` using various subdirectory such as `images`, `stylesheets`, etc.

**Reply by maelle** [reply]
Merci ! Does this magically get recognized as resource?

**Reply by mcanouil** [reply]
Only if it is used as resources under keys such as `css`, `logo`, `theme`, ...
If you don't want to worry and include everything (possibly multiple times):

```yml
resources:
  - /assets
```

**Comment by younes-io** [comment]
I struggle to make this work for a blog. I got  the image `assets/background.png`

The browser makes these requests:
```
 /blog/2023-12-15-article/background.png (404: Not Found)
 /blog/background.png (404: Not Found)
```

It does so because the image is defined in the html like this 

` style="background-image: url(background.png), url(../background.png);`

**Reply by younes-io** [reply]
BTW, I defined it this way to make sure each article has its own background image..

**Reply by mcanouil** [reply]
How do you expect Quarto or the browser to find the image in assets if you don't tell it is in assets in the first place?
I might be missing something since you did not provide any concrete reproducible example.

**Reply by younes-io** [reply]
It works now, I had to do this

```
  resources:
    - /assets
    - /blog/**/*.jpg
```


It didn't "see" the images in `blog/articles/` so when I reference them in the article, they're not found..
If you have a "cleaner" way of doing it, I'm all ears

**Reply by mcanouil** [reply]
I can't give you alternatives since you did not share what you did. You only shared partial pieces.
